tests/qa_automation/prepare_qa_repo.pm

```diff
@@ -0,0 +1,44 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: the step to prepare QA:Head repository
+# Maintainer: Yong Sun <yosun@suse.com>
+
+use strict;
+use testapi;
+use utils;
+use base "opensusebasetest";
+
+# Add qa head repo for kernel testing. If QA_HEAD_REPO is set,
+# remove all existing zypper repos first
+sub prepare_repos {
+    my $self         = shift;
+    my $qa_head_repo = get_required_var('QA_HEAD_REPO', '');
+    my $qa_web_repo  = get_var('QA_WEB_REPO', '');
+
+    zypper_call("--no-gpg-check ar -f $qa_head_repo qa-ibs");
+    if ($qa_web_repo) {
+        zypper_call("--no-gpg-check ar -f $qa_web_repo qa-web");
+    }
```

Comment:
Just a small remark here, you could replace your `if (...) { zypper_call... }` by `zypper_call(...) if (...);` ;-)

Reply:
easy read for Perl none expert, I would like to keep :)

Reply:
I'm neither a Perl expert and I personally find the other more readable :-) And a lot of openQA test codes uses it.
But it's ok for me ;-)